spec/shared_contexts/digests.rb

```diff
@@ -17,7 +17,12 @@
 
   def self.with_digest_algorithms(&block)
     DIGEST_ALGORITHMS_TO_TRY.each do |digest_algorithm|
-      describe("when digest_algorithm is #{digest_algorithm}", :digest_algorithm => digest_algorithm) do
+      describe(
+        "when digest_algorithm is #{digest_algorithm}",
+        {:digest_algorithm => digest_algorithm}.merge(
+          Puppet::Util::Platform.windows? && ['sha384', 'sha512'].include?(digest_algorithm) ? {:skip => "Windows Long File Name support is incomplete PUP-8257"} : {}
```

Comment:
I think Jorie and I have found a way to make this way less hacky. We'll work on updating the PR with what we've found.